### PR TITLE
[Tests]pausing_test.go:Reduces API calls during the test

### PR DESCRIPTION
**Pull request by Barakmor1**
Instead of Updating VM runStrategy in pausing_test.go create vm with with the desired runStrategy.

### What this PR does
Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by fossedihelm — APPROVED** [review]

File: tests/pausing_test.go
Comment (xpivarc) [inline tests/pausing_test.go:209]: The namespace should not be used.

**Review by EdDev — CHANGES_REQUESTED** [review]
See @xpivarc feedback.

**Comment by xpivarc** [comment]
Merge conflict, interesting this should not happen...

**Comment by EdDev** [comment]
> Merge conflict, interesting this should not happen...

Not that surprising to see an XXL PR [1] causing a large amount of PRs conflicts.
So large PRs not only is hard to review and suffers from a lot of potential conflicts (e.g. this one could get in earlier and cause a rebase on the bigger one), it also makes it hard for other PRs that are posted.

[1] https://github.com/kubevirt/kubevirt/pull/12346

**Comment by fossedihelm** [comment]
> > Merge conflict, interesting this should not happen...
> 
> Not that surprising to see an XXL PR [1] causing a large amount of PRs conflicts. So large PRs not only is hard to review and suffers from a lot of potential conflicts (e.g. this one could get in earlier and cause a rebase on the bigger one), it also makes it hard for other PRs that are posted.
> 
> [1] #12346

It's not the PR itself. It's pretty clear that there is one commit that moves a pkg widely used.
Even if it was a single commit PR the outcome would have been the same.

**Comment by xpivarc** [comment]
/hold
We should deprecate this function first if nobody is committing to remove this as whole. This will come with explicit approval that we want to do something like this and it will make sure we converge.

**Comment by EdDev** [comment]
> It's not the PR itself. It's pretty clear that there is one commit that moves a pkg widely used.
> Even if it was a single commit PR the outcome would have been the same.

The point is that a big PR and/or commit broke many other PRs.
And it is symetric, if this PR would have got in, it would have collided slightly with that big PR and required a rebase there.

For me, that means that small changes are better than large changes. And even if it does make sense to group them in one PR, it cannot be forced on the author.

> /hold
> We should deprecate this function first if nobody is committing to remove this as whole. This will come with explicit approval that we want to do something like this and it will make sure we converge.

This is unfortunate to see work being blocked on such reasoning.
While some encourage small steps with small changes, others block it in favor of large changes without leaving freedom for others to approve/review this.

**Comment by Barakmor1** [comment]
I don’t believe this is a strong reason to hold up the PR. However, here is the PR for the deprecation:
https://github.com/kubevirt/kubevirt/pull/12572

/unhold

File: tests/pausing_test.go
Comment (Barakmor1) [reply tests/pausing_test.go]: Done.

**Review by EdDev — APPROVED** [review]

File: tests/pausing_test.go
Comment (xpivarc) [inline tests/pausing_test.go:201]: nit: a0f8aae0cbb05013d636aed602076514df0e3f39 , suggestion for commit `Update tests to use correct RunStrategy instead of creating VMs stopped and then starting them. This simplifies these test cases.`

`Instead of Updating VM runStrategy in pausing_test.go create vm with
with the desired runStrategy.` is seems cryptic, at least for me.

File: tests/pausing_test.go
Comment (xpivarc) [inline tests/pausing_test.go:207]: Do we have any convenient and safe function that would create a VM and expect it to start? What I mean by the "safe" is to use libwait to also look for any warnings/errors in the startup flow.

File: tests/pausing_test.go
Comment (xpivarc) [inline tests/pausing_test.go:212]: How come this happens?  The `vm, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, v12.CreateOptions{})` should populate the namespace.

File: tests/pausing_test.go
Comment (Barakmor1) [reply tests/pausing_test.go:207]: Not sure if we have this function. Please let me know if you come across it. Thanks!

File: tests/pausing_test.go
Comment (Barakmor1) [reply tests/pausing_test.go]: Done, thanks for the feedback.

File: tests/pausing_test.go
Comment (Barakmor1) [reply tests/pausing_test.go]: You are right :) 
I removed this commit.

**Review by nunnatsa — APPROVED** [review]

**Review by xpivarc — COMMENTED** [review]
/approve

File: tests/pausing_test.go
Comment (xpivarc) [inline tests/pausing_test.go:142]: This shadows the `vm` variable in the `Context`. This needs to be fixed

Comment (Barakmor1) [reply tests/pausing_test.go]: Done :)

**Comment by Barakmor1** [comment]
@xpivarc PTAL

File: tests/pausing_test.go
Comment (xpivarc) [inline tests/pausing_test.go:136]: Please don't use "global" errors. It is simple to mess up.

**Review by xpivarc — COMMENTED** [review]
/approve
/hold cancel

File: tests/pausing_test.go
Comment (xpivarc) [reply tests/pausing_test.go:136]: @Barakmor1 Please address this before merging

Comment (Barakmor1) [reply tests/pausing_test.go]: Done

**Comment by brianmcarey** [comment]
/hold

The test lane failures look to be related.

**Comment by Barakmor1** [comment]
@brianmcarey Thanks for letting me know, i think it should be good now.

/unhold

**Comment by Barakmor1** [comment]
@brianmcarey Thanks for letting me know, i think it should be good now.

/unhold

**Review by orelmisan — APPROVED** [review]
Thank you @Barakmor1

**Pull request merged**
